lib/connectionConfig/iothubConnectionConfig.ts

```diff
@@ -51,9 +52,9 @@ export namespace IotHubConnectionConfig {
    * @param {string} [path]           - The name/path of the entity (hub name) to which the connection needs to happen
    */
   export function create(connectionString: string, path?: string): IotHubConnectionConfig {
-    if (!connectionString || (connectionString && typeof connectionString !== "string")) {
-      throw new Error("'connectionString' is a required parameter and must be of type: 'string'.");
-    }
+    throwTypeErrorIfParameterMissing("connectionString", connectionString);
+    connectionString = String(connectionString);
```

Comment:
We can skip the parameter missing check here.
If none was provided, connectionString will be `undefined` and `parseConnectionString` will give appropriate errors